### PR TITLE
NN-4170 restructure and get tests working with basics of new flow

### PR DESCRIPTION
**Pull request by timoj58**
this PR includes
* copy all details files to new incidentRole and rename
* amend tests to test roles related things
* add new paths, refactor code to use adjudication number
* remove creation page
* new api added (no server side)

### CODE REVIEW

File: server/routes/incidentRole/incidentRole.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Request, Response } from 'express'
+import PlaceOnReportService from '../../services/placeOnReportService'
+import IncidentRolePage, { PageRequestType } from './incidentRolePage'
+
+export default class IncidentRoleRoutes {
+  page: IncidentRolePage
+
+  constructor(placeOnReportService: PlaceOnReportService) {
+    this.page = new IncidentRolePage(PageRequestType.CREATION, placeOnReportService)
```
Comment (sp-watson) [inline server/routes/incidentRole/incidentRole.ts:9]: No need for creation

File: server/routes/incidentRole/incidentRoleValidation.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { FormError } from '../../@types/template'
```
Comment (sp-watson) [inline server/routes/incidentRole/incidentRoleValidation.ts:1]: Not sure this needs duplicating, but might do.. i need to check..

If it already exists then a rename is always better (if possible) as we keep the history then

File: server/routes/incidentRole/incidentRoleSearchValidation.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { FormError } from '../../@types/template'
```
Comment (sp-watson) [inline server/routes/incidentRole/incidentRoleSearchValidation.ts:1]: As per other validation.. Not sure this needs duplicating

File: server/routes/incidentRole/incidentRolePage.ts [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+  saveToApiUpdate = async (
+    draftId: number,
+    data: IncidentDetails,
+    removeExistingOffences: boolean,
+    currentUser: User,
+    existingAdjudication: DraftAdjudicationResult
+  ): Promise<DraftAdjudicationResult> => {
+    // eslint-disable-next-line no-return-await
+    // TODO merge in the current values for location and date...
+    return this.placeOnReportService.editDraftIncidentDetails(
```
Comment (sp-watson) [inline server/routes/incidentRole/incidentRolePage.ts:260]: We need a new API call for this

File: server/routes/incidentRole/incidentRoleValidation.ts (hunk 2 above)
Comment (timoj58) [reply server/routes/incidentRole/incidentRoleValidation.ts:1]: so for this one we do need two.  As this contains the validation for roles only, and the details version will have the roles part deleted once we remove that from the page

File: server/@types/express/index.d.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ declare module 'express-session' {
     returnTo: string
     nowInMinutes: number
     redirectUrl: string
+    prisonerNumber: string
```
[the remark this answers is not in the document]
Comment (timoj58) [reply server/@types/express/index.d.ts:15]: removed

File: helm_deploy/values-dev.yaml [hunk 6]
```diff
@@ -19,6 +19,7 @@ generic-service:
     PRISONER_SEARCH_API_URL: https://prisoner-offender-search-dev.prison.service.justice.gov.uk
     ADJUDICATIONS_URL: http://hmpps-manage-adjudications
     GOTENBERG_API_URL: http://hmpps-manage-adjudications-gotenberg
+    YOI_NEW_PAGES_FEATURE_FLAG: true
```
[the remark this answers is not in the document]
Comment (timoj58) [reply helm_deploy/values-dev.yaml:22]: i thought we want it on in all envs other than prod, which is what i did?

**Review by sp-watson — DISMISSED** [review]

**Review by sp-watson — APPROVED** [review]

**Pull request merged**